selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/client/keycloak/executor/OpenShiftPodCommandExecutor.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core.client.keycloak.executor;
+
+import static java.lang.String.format;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.che.selenium.core.provider.OpenShiftWebConsoleUrlProvider;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgent;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgentException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** @author Dmytro Nochevnov */
+@Singleton
+public class OpenShiftPodCommandExecutor implements KeycloakCommandExecutor {
+  private static final Logger LOG = LoggerFactory.getLogger(OpenShiftPodCommandExecutor.class);
+
+  private static final boolean IS_MAC_OS =
+      System.getProperty("os.name").toLowerCase().startsWith("mac");
+  private static final String DEFAULT_OPENSHIFT_USERNAME = "developer";
+  private static final String DEFAULT_OPENSHIFT_PASSWORD = "any";
```

Comment:
You may need a token since if cluster is configured  with oAuth, there's no way you can do `oc login -u -p`. Can we introduce a token for such use cases? Say, before running a test, you obtain a token, export it as env (or save to configuration file) and then run tests.

Reply:
Thank you for notice. Which environment variable we can use to hold the token?

Reply:
You may pick any reasonable name. And if set, then oc login should use it. If not, username and password are used (default unless set by a user).

You can get your token this way:
```
oc login -u developer -p password
oc whoami -t
```

My point is that even though this PR makes interaction with Keycloak more efficient, it won't help if you need to run tests against any infra that different from a default cluster started with `oc cluster up`

Reply:
I can't find parameter to pass auth token to "oc login" command [in docs](https://docs.openshift.com/enterprise/3.0/dev_guide/authentication.html). 
Did you mean that we should add auth token directly into the ~/.kube/config file?

Reply:
```
eugene@ivantsoft ~/projects/rhel-stacks/che-assembly (master) $ oc login --help
Log in to your server and save login for subsequent use 

First-time users of the client should run this command to connect to a server, establish an authenticated session, and
save connection to the configuration file. The default configuration will be saved to your home directory under
".kube/config". 

The information required to login -- like username and password, a session token, or the server details -- can be
provided through flags. If not provided, the command will prompt for user input as needed.

Usage:
  oc login [URL] [options]

Examples:
  # Log in interactively
  oc login
  
  # Log in to the given server with the given certificate authority file
  oc login localhost:8443 --certificate-authority=/path/to/cert.crt
  
  # Log in to the given server with the given credentials (will not prompt interactively)
  oc login localhost:8443 --username=myuser --password=mypass

Options:
  -p, --password='': Password, will prompt if not provided
  -u, --username='': Username, will prompt if not provided      --certificate-authority='': Path to a cert file for the
certificate authority
      --insecure-skip-tls-verify=false: If true, the server's certificate will not be checked for validity. This will
make your HTTPS connections insecure
      --token='': Bearer token for authentication to the API server

Use "oc options" for a list of global command-line options (applies to all commands).
eugene@ivantsoft ~/projects/rhel-stacks/che-assembly (master) $ 

```

Reply:
Thank you, Eugene.